probe/probe.go

```diff
@@ -487,19 +487,21 @@ func getKubeArmorContainers(c *k8s.Client, o Options) map[string]*KubeArmorPodSp
 // ProbeRunningKubeArmorNodes extracts data from running KubeArmor daemonset  by executing into the container and reading /tmp/kubearmor.cfg
 func ProbeRunningKubeArmorNodes(c *k8s.Client, o Options) ([]KubeArmorProbeData, map[string]KubeArmorProbeData, error) {
 	// KubeArmor Nodes
-	nodes, err := c.K8sClientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		return []KubeArmorProbeData{}, nil, fmt.Errorf("error occured when getting nodes %s", err.Error())
-	}
+	pods, err := c.K8sClientset.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+		LabelSelector: "kubearmor-app=kubearmor",
+	})
 
-	if len(nodes.Items) == 0 {
+	if err != nil || len(pods.Items) == 0 {
 		return []KubeArmorProbeData{}, nil, fmt.Errorf("no nodes found")
 	}
 	nodeData := make(map[string]KubeArmorProbeData)
 
 	var dataList []KubeArmorProbeData
-	for i, item := range nodes.Items {
-		data, err := readDataFromKubeArmor(c, o, item.Name)
+	for i, item := range pods.Items {
+		if item.Status.Phase != corev1.PodRunning {
+			continue
+		}
+		data, err := readDataFromKubeArmor(c, item)
 		if err != nil {
 			return []KubeArmorProbeData{}, nil, err
```

Comment:
An error for any one of the nodes will do an early return here. We won't get probe output of other nodes. Is it possible to handle this?
![image](https://github.com/kubearmor/kubearmor-client/assets/54525605/a01ec6b0-9e5e-40e5-b507-f5a467da3993)